src/EditorFeatures/Core/Implementation/IntelliSense/AsyncCompletion/CompletionSource.cs

```diff
@@ -473,5 +498,141 @@ private ImmutableArray<AsyncCompletionData.CompletionFilter> GetFilters(RoslynCo
 
             return current == questionPosition;
         }
+
+        /// <summary>
+        /// Provides an efficient way to compute a set of completion filters associated with a collection of completion items.
+        /// Presence of expander and filter in the set have different meanings. Set contains a filter means the filter is
+        /// available but unselected, whereas it means available and selected for an expander. Note that even though VS supports 
+        /// having multiple expanders, we only support one.
+        /// </summary>
+        private class FilterSet
+        {
+            // Cache all the VS completion filters which essentially make them singletons.
+            // Because all items that should be filtered using the same filter button must 
+            // use the same reference to the instance of CompletionFilter.
+            private static readonly Dictionary<string, AsyncCompletionData.CompletionFilter> s_filterCache =
+                new Dictionary<string, AsyncCompletionData.CompletionFilter>();
+
+            private BitVector32 _vector;
+            private static readonly ImmutableArray<int> s_filterMasks;
+            private static readonly int s_expanderMask;
+            private static AsyncCompletionData.CompletionExpander _expander = null;
+
+            public static ImmutableArray<CompletionItemFilter> Filters => CompletionItemFilter.AllFilters;
+
+            public static AsyncCompletionData.CompletionExpander Expander
+            {
+                get
+                {
+                    if (_expander == null)
+                    {
+                        var addImageId = Shared.Extensions.GlyphExtensions.GetImageCatalogImageId(KnownImageIds.ExpandScope);
+                        _expander = new AsyncCompletionData.CompletionExpander(
+                            EditorFeaturesResources.Expander_display_text,
+                            accessKey: "a",
```

Comment:
in the PR description you said that Alt+X is the shortcut

Reply:
Alt + X is already used, so I switched to `A` instead. But I forgot to update the description.